src/modules/localVersionCoordination/LocalVersionCoordinationPrompt.tsx

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect, useState } from 'react';
+import ConfirmationDialog from '@/components/elements/ConfirmationDialog';
+import useLocalVersionCoordination, {
+  LocalVersionedRecordType,
+} from '@/modules/localVersionCoordination/hooks/useLocalVersionCoordination';
+import { cache } from '@/providers/apolloClient';
+
+interface Props {
+  recordType: LocalVersionedRecordType;
+  recordId: string;
+  currentVersion: number;
+  onReload?: VoidFunction;
+}
+const LocalVersionCoordinationPrompt: React.FC<Props> = ({
+  recordType,
+  recordId,
+  currentVersion,
+  onReload,
+}) => {
+  const [acknowledgedVersion, setAcknowledgedVersion] =
+    useState(currentVersion);
+  const [loading, setLoading] = useState(false);
+  const latestVersion = useLocalVersionCoordination(
+    recordType,
+    recordId,
+    currentVersion
+  );
+
+  const handleReload = useCallback(() => {
+    setLoading(true);
+    if (onReload) {
+      onReload();
+    } else {
+      // default: on reload evict the cache
+      cache.evict({ id: `${recordType}:${recordId}` });
+    }
+  }, [onReload, recordId, recordType]);
+
+  useEffect(() => {
+    if (currentVersion === latestVersion) {
+      setLoading(false);
+    }
+  }, [currentVersion, latestVersion]);
+
+  const handleContinue = () => {
+    setAcknowledgedVersion(latestVersion);
+  };
+
+  // hide prompt if we are on the latest version OR we are on an old version but the user has acknowledged it
+  const showPrompt =
+    latestVersion > Math.max(acknowledgedVersion, currentVersion);
```

Comment:
perhaps we should show a snackbar or alert even if the user has acknowledge the stale record